src/main/java/xades4j/production/SignerBES.java

```diff
@@ -260,6 +277,17 @@ public final XadesSignatureResult sign(
                 throw new UnsupportedAlgorithmException(
                         "Digest algorithm not supported in the XML Signature provider",
                         digestAlgUri, ex);
+            } catch (InvalidCanonicalizerException ex)

```

Comment:
Great suggestion. Based on your feedback, I re-introduced CanonicalizerUtils class. For now, it only has the "HACK" approach defined. I hope this was your interpretation. Note that, I do not perform a check for null. See commit https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762 and https://github.com/luisgoncalves/xades4j/pull/111/commits/c6bd19eff6eea4238918bf84d4ecb64c2bd59762#diff-d020145c5aef76d0f8394ec75d621dfa.